crates/services/producer/src/block_producer/gas_price.rs

```diff
@@ -0,0 +1,51 @@
+use fuel_core_types::fuel_types::BlockHeight;
+
+/// The parameters required to retrieve the gas price for a block
+pub struct GasPriceParams {
+    block_height: BlockHeight,
+}
+
+impl GasPriceParams {
+    /// Create a new `GasPriceParams` instance
+    pub fn new(block_height: BlockHeight) -> Self {
+        Self { block_height }
+    }
+
+    pub fn block_height(&self) -> BlockHeight {
+        self.block_height
+    }
+}
+
+impl From<BlockHeight> for GasPriceParams {
+    fn from(block_height: BlockHeight) -> Self {
+        Self { block_height }
+    }
+}
+
+/// Interface for retrieving the gas price for a block
+pub trait ProducerGasPrice {
+    /// The gas price for all transactions in the block.
+    fn gas_price(&self, params: GasPriceParams) -> Option<u64>;
+}
+
+pub struct MockProducerGasPrice {
```

Comment:
Maybe we can use `mockall` instead?

Reply:
I dislike `mockall`.

We have a lot of tests in our codebase that test the internals of our code rather the externals. We do that with `mockall`'s `expect_`, `with`, `times`, and similar methods. It's true that `mockall` would give us a struct that is basically identical to what I added, but it also gives us a bunch of things that are used to write bad tests, so I avoid it.

Reply:
If you don't like these functions, you can not use them=)

If you have a nice tool with some feature that you don't need, simply don't use them=) Auto-generation structures for testing simplify supporting of this code in the future.

If you worry that in the future someone will use `expect` and other methods, then usage of the custom structure will not solve that=D The person simply will use `mockall` and will use `expect` because he wants to=)